### PR TITLE
[Test-Proxy] eng/common templates

### PR DESCRIPTION
**Pull request by scbedd**
Changes in this PR:
- Add test proxy tool and docker invocations to the eng/common yaml.
- Update eng/common with the SSL-related items so they can be trusted by language-specific tooling
- Update documentation and build to reflect new certificate files location.

Ok. @weshaggard I have added a `prepare.ps1` to the `test-proxy/docker` folder. Updated the documentation that refers to the certificate by name/location. Updated the docker build to be able to run the `prepare.ps1` script. 

[Successful Test invocation using these location changes in sdk-for-python](https://dev.azure.com/azure-sdk/playground/_build/results?buildId=1050334&view=results)
[Successful Test container publish](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1050428&view=logs&j=c370c9e0-ef2e-5a74-495e-4719e52f70ec&t=21eced4c-c8be-5dcb-8a86-9b163bc55754)

### CODE REVIEW

File: eng/common/testproxy/test-proxy-tool.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+parameters:
+  rootFolder: '$(Build.SourcesDirectory)'
+
+steps:
+  - pwsh: |
+        . $(Build.SourcesDirectory)/eng/common/scripts/common.ps1
+
+        if ($FindArtifactForApiReviewFn -and (Test-Path "Function:$FindArtifactForApiReviewFn"))
+        {
+            &$FindArtifactForApiReviewFn
```
Comment (weshaggard) [inline eng/common/testproxy/test-proxy-tool.yml:10]: How ds this script have anything to do with loading the certificate?

**Comment by benbp** [comment]
@scbedd one possible solution: you could publish another docker image from tools that just contains the dev cert and the desired path, and then use multiple `FROM` directives to overlay your "test proxy assets" filesystem into the image build. That way you wouldn't need any external directory context, and could manage updates to the dev cert easily.

File: eng/common/testproxy/test-proxy-tool.yml (hunk 1 above)
Comment (scbedd) [reply eng/common/testproxy/test-proxy-tool.yml:10]: I didn't update this appropriately from my test source :| 

Thanks for the catch 👍

**Comment by scbedd** [comment]
@benbp I am going to head this direction. First thing I'm going to do is force the build of the current dockerfile images from the root of the repo, thereby allowing access to the common assets. After that, I'll add two docker images that abstract the build of those assets. I would do it now, but I need to get this merged quickly.

File: eng/common/testproxy/test-proxy-docker.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+parameters:
+    rootFolder: '$(Build.SourcesDirectory)'
+
+steps:
+  - pwsh: |
+        . $(Build.SourcesDirectory)/eng/common/scripts/common.ps1
```
Comment (weshaggard) [inline eng/common/testproxy/test-proxy-docker.yml:6]: These lines might be worth adding an script to execute. Especially if you plan to have devs do these steps locally.

Comment (scbedd) [reply eng/common/testproxy/test-proxy-docker.yml:6]: I'm seeing an integration issue that's going to force this, so good point!

File: eng/common/pipelines/templates/steps/install-pipeline-generation.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ steps:
       dotnet tool install
       Azure.Sdk.Tools.PipelineGenerator
       --version 1.0.2-dev.20210621.4
-      --add-source https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk/nuget/v3/index.json
+      --add-source https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json
```
Comment (scbedd) [inline eng/common/pipelines/templates/steps/install-pipeline-generation.yml:13]:
```suggestion
      --add-source https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk/nuget/v3/index.json
```

Comment (scbedd) [reply eng/common/pipelines/templates/steps/install-pipeline-generation.yml:13]: Reverting this to what it was before. There is no reason for me to touch this.

**Pull request merged**